pkg/controller/networkpolicy_reconciler.go

```diff
@@ -336,12 +346,33 @@ func (reconciler *NetworkpolicyReconciler) updateDefaultNetworkPolicy(namespace
 }
 
 func (reconciler *NetworkpolicyReconciler) createDefaultNetworkPolicy(namespace string, networkpolicyName string) error {
-	desiredPolicy := reconciler.defaultNetworkPolicies[networkpolicyName].DeepCopy()
-	desiredPolicy.ObjectMeta.Namespace = namespace
+	if _, ok := reconciler.defaultNetworkPolicies[networkpolicyName]; !ok {
+		klog.Infof("Network policy not in buffer. Load %s", networkpolicyName)
 
-	if _, err := reconciler.kubeclientset.NetworkingV1().NetworkPolicies(namespace).Create(desiredPolicy); err != nil {
-		return err
+		networkPolicyConfigmap, err := reconciler.kubeclientset.CoreV1().ConfigMaps("kube-system").Get(networkpolicyName, meta_v1.GetOptions{})
+		if err != nil {
+			klog.Errorf("Failed to get network policy %s", networkpolicyName)
+			return fmt.Errorf("Failed to get network policy %s", networkpolicyName)
+		}
+
+		var policy networkingv1.NetworkPolicy
+		if err := yaml.Unmarshal([]byte(networkPolicyConfigmap.Data["policy"]), &policy); err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to unmarshal default network policy configmap ('%s:%s') into network policy object: %v\n", namespace, networkpolicyName, err)
+		}
+		reconciler.defaultNetworkPolicies[networkpolicyName] = &policy
+		klog.Infof("Network policy %s loaded into buffer. New Buffer length: %v", policy.GetName(), len(reconciler.defaultNetworkPolicies))
 	}
-	return nil
 
+	if _, ok := reconciler.defaultNetworkPolicies[networkpolicyName]; !ok {
+		klog.Errorf("Network not found in buffer after load %s", networkpolicyName)
```

Comment:
Shouldn't an error be returned after the log? Currently, this case will return `nil`.